main.go

```diff
@@ -135,42 +135,31 @@ func extractText(n *html.Node) string {
 }
 
 // TODO: Clean up and improve this code.
-// GitHub Flavored Markdown fenced code block with highlighting.
+// BlockCode renders GitHub Flavored Markdown fenced code block with highlighting.
 func (*renderer) BlockCode(out *bytes.Buffer, text []byte, lang string) {
 	doubleSpace(out)
 
 	// parse out the language name
-	count := 0
-	for _, elt := range strings.Fields(lang) {
-		if elt[0] == '.' {
-			elt = elt[1:]
-		}
-		if len(elt) == 0 {
-			continue
-		}
-		out.WriteString(`<div class="highlight highlight-`)
-		attrEscape(out, []byte(elt))
-		lang = elt
-		out.WriteString(`"><pre>`)
-		count++
-		break
+	lang = strings.TrimSpace(lang)
+	if len(lang) > 0 && lang[0] == '.' {
+		lang = lang[1:]
```

Comment:
> Actually, I didn't find any example of usage, when `for _, elt := range strings.Fields(lang) { ... }` loop is useful. That's why I deleted it. If I'm not right, please explain me, which situations this loop manages.

The original code for `BlockCode` is largely taken from `blackfriday.Html.BlockCode`, with minor GFM-specific modifications applied. That includes the `for _, elt := strings.Fields(lang)` loop. You can see it upstream at:

https://github.com/russross/blackfriday/blob/6d1ef893fcb01b4f50cb6e57ed7df3e2e627b6b2/html.go#L258-L287

Reply:
Okay, I will revert it. But anyway I can't understand, where it could be useful.

Reply:
> But anyway I can't understand, where it could be useful.

In that case, I think a good next step is to open an issue upstream in blackfriday and discuss it there. If we can clarify or improve the implementation there, it'll be trivial to backport the improvements here.